### ISSUE TITLE
Inaccurate query being used to fetch mid servers

### ISSUE THREAD

**Issue by mitchell852** [apache/trafficcontrol#3472]
I don't believe this query used in this function is quite right:

https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/server/servers.go#L303

It basically evaluates to this:

```
SELECT *
FROM server AS s
WHERE s.id IN (
    SELECT mid.id FROM server mid
    JOIN cachegroup cg ON cg.id IN (
        SELECT cg.parent_cachegroup_id
        FROM server s
        JOIN cachegroup cg ON cg.id = s.cachegroup
        WHERE s.id IN (1,2,3))
    JOIN type t ON mid.type = (SELECT id FROM type WHERE name = 'MID'))
```

which regardless of which server id is fed into it always seems to return ALL the mid servers...

I think it needs to be like this:

```
SELECT *
FROM server AS s 
JOIN type AS t ON s.type = t.id
WHERE t.name = 'MID' AND s.cachegroup IN (
    SELECT cg.parent_cachegroup_id FROM cachegroup AS cg 
    WHERE cg.id IN (
        SELECT s.cachegroup FROM server AS s 
        WHERE s.id IN (1,2,3)))
```

This does these things to fetch the mids associated with a list of edges:

1. gets the cache groups of the edges
2. get the parent cachegroups for those cachegroups (found in 1)
3. get the servers that belong to those  parent cachegroups (found in 2) that are mids

Basically the query is meant to answer the question "if my ds uses these edges, which mids does my ds use based on the relationship of the cache groups"
